community/cypher/cypher/src/test/scala/org/neo4j/cypher/MutatingIntegrationTest.scala

```diff
@@ -478,4 +478,14 @@ return distinct center""")
 
     assertStats(result, nodesCreated = 1)
   }
+
+  test("should be possible to remove nodes created in the same query") {
+    val result = execute(
+      """CREATE (a)-[:FOO]->(b)
+         WITH a, b
```

Comment:
Yes. It does not make any semantic difference but syntax requires `WITH` between `CREATE` and `MATCH`